src/backend/distributed/planner/multi_explain.c

```diff
@@ -99,10 +173,655 @@ MultiExplainOneQuery(Query *query, IntoClause *into, ExplainState *es,
 		appendStringInfo(es->str, "%s\n", formattedPlanString);
 	}
 
-	/* if explain printing isn't enabled, print error only after planning */
-	if (!ExplainMultiLogicalPlan && !ExplainMultiPhysicalPlan)
+	if (!ExplainDistributedQueries)
 	{
 		appendStringInfo(es->str, "explain statements for distributed queries ");
+		appendStringInfo(es->str, "are not enabled\n");
+		return;
+	}
+
+	workerJob = multiPlan->workerJob;
+
+	if (list_length(workerJob->dependedJobList) > 0)
+	{
+		appendStringInfo(es->str, "explain statements for complex queries ");
 		appendStringInfo(es->str, "are currently unsupported\n");
```

Comment:
Yes. It's a bit tricky because the queries use intermediate tables that are only created during re-partitioning, but we don't want to actually run the re-partitioning job when doing a regular EXPLAIN, so we'd need some infrastructure for creating the tables. There are no fundamental limitations preventing us from doing that. We originally marked that as "phase 2" of EXPLAIN, with the most critical parts (this PR) being phase 1. 

I suppose we can show something, but it doesn't seem like it would convey very useful information.